data-processor/src/test/groovy/io/micronaut/data/model/query/builder/SqlQueryBuilderSpec.groovy

```diff
@@ -173,6 +173,50 @@ interface MyRepository {
 
     }
 
+    void "test h2 crud"() {
+        given:
+        def annotationMetadata = buildTypeAnnotationMetadata('''
+package test;
+import io.micronaut.data.annotation.*;
+import io.micronaut.data.model.query.builder.sql.*;
+import java.lang.annotation.*;
+import io.micronaut.data.jdbc.annotation.*;
+import io.micronaut.context.annotation.*;
+
+@MyAnnotation(dialect = Dialect.H2)
+interface MyRepository {
+}
+
+@RepositoryConfiguration(
+        queryBuilder = SqlQueryBuilder.class
+)
+@SqlQueryConfiguration(
+    @SqlQueryConfiguration.DialectConfiguration(
+        dialect = Dialect.H2,
+        positionalParameterFormat = "$%s",
+        escapeQueries = false
+    )
+)
+@Retention(RetentionPolicy.RUNTIME)
+@Repository
+@interface MyAnnotation {
+    @AliasFor(annotation = Repository.class, member = "dialect")
+    Dialect dialect() default Dialect.ANSI;
+}
+''')
+
+        PersistentEntity entity = PersistentEntity.of(Sale)
+        QueryBuilder builder = new SqlQueryBuilder(Dialect.H2)
+        def queryModel = QueryModel.from(entity).eq("name", QueryParameter.of("name"))
+
+        expect:
+        builder.dialect == Dialect.H2
```

Comment:
I think `buildQuery` method has been changed in 4.0 and need `AnnotationMetadata` so just need to add this `AnnotationMetadata.EMPTY_METADATA` as first parameter, for example.

Reply:
@radovanradic I fixed failing test